### PR TITLE
Stats: Modify products upgrade link according to the "is_commercial" classifier on the Add-Ons page

### PR DESCRIPTION
**Pull request by dognose24**
Related to #80258 

## Proposed Changes

* Introduce the `is_commercial` option to determine the Add-On card for Stats products.
* Navigate to the respective purchase page according to the `Add-Ons` card product.

## Testing Instructions

* Spin up the change with the Calypso Live link.
* Navigate to the `Add-Ons` page for a Simple site with the feature flag: `/add-ons/{site-slug}?flags=stats/paid-wpcom-stats`.
* Ensure the `Add-Ons` product card displays according to the site's `is_commercial` status.
* Ensure the `Upgrade Stats` button directs to the correct purchase page for Stats PWYW and Commercial products.

|Non-commercial site|Commercial site|
|-|-|
|截圖 2023-09-06 下午12 11 15|截圖 2023-09-06 下午12 10 14|

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/add-ons/hooks/use-add-ons.ts [hunk 1]
```diff
@@ -159,15 +175,37 @@ const useAddOns = ( siteId?: number ): ( AddOnMeta | null )[] => {
...
+				if (
+					isSiteMarkedCommercial === true &&
+					PRODUCT_JETPACK_STATS_PWYW_YEARLY === addOn.productSlug
+				) {
+					return false;
+				}
+
+				// Hide Stats Commercial add-on if the site is not marked as commercial.
+				if (
+					isSiteMarkedCommercial !== true &&
```
Comment (kangzj) [inline client/my-sites/add-ons/hooks/use-add-ons.ts]: Could we please show both cards when `isSiteMarkedCommercial` is `null`? because chances are in the coming few weeks, most sites will still not be identified yet.

File: client/my-sites/add-ons/hooks/use-add-ons.ts [hunk 2]
```diff
@@ -140,6 +153,9 @@ const useAddOns = ( siteId?: number ): ( AddOnMeta | null )[] => {
 			}
 		}
 
+		// Determine which Stats Add-On to show based on the site's commercial classification.
+		const isSiteMarkedCommercial = getSiteOption( state, siteId, 'is_commercial' );
```
Comment (kangzj) [inline client/my-sites/add-ons/hooks/use-add-ons.ts:157]: The page flickers on load, but it seems not related to the PR.


https://github.com/Automattic/wp-calypso/assets/1425433/46866ea2-a8cc-4619-943d-0c7c657fae7d

File: client/my-sites/add-ons/hooks/use-add-ons.ts (hunk 1 above)
Comment (kangzj) [reply client/my-sites/add-ons/hooks/use-add-ons.ts]: <strike> Actually we could possibly defer the determination to the purchase page, couldn't we? </strike>

Comment (dognose24) [reply client/my-sites/add-ons/hooks/use-add-ons.ts]: Addressed in https://github.com/Automattic/wp-calypso/pull/81417/commits/aa147fd8bd812295510239d632c04dbb2aacfb5e.
